visual-js/visual-wdio/src/SauceVisualService.ts

```diff
@@ -404,9 +408,19 @@ export default class SauceVisualService implements Services.ServiceInstance {
 
       const sessionId = browser.sessionId;
       const jobId = (browser.capabilities as any)['jobUuid'] || sessionId;
+
+      let clipSelector = options.clipSelector ?? this.clipSelector;
+      if (clipSelector) {
+        const element = await browser.$(clipSelector);
+        clipSelector = element.elementId;
+      }
```

Comment:
Perhaps this instead? IMO I'd try not to mutate variables and reassign them to a value that doesn't match the name  (`clipElement` in a `clipSelector`)

```
const clipSelector = options.clipSelector ?? this.clipSelector;
const clipElement = clipSelector ? await browser.$(clipSelector) : null;
```